core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -353,4 +353,101 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
       .recordStream(pollTimeout)
       .evalMap(r => process(r) <* consumer.commitSync(r.nextOffset))
 
+  /** Returns a stream that processes records using the specified function,
+    * committing offsets for successfully processed records, either after
+    * processing the specified number of records, or after the specified time
+    * has elapsed since the last offset commit. If the processing function
+    * returns a failure, offsets for records successfully processed before that
+    * failure will be committed, and then the stream will halt with that
+    * failure. This is at-least-once processing: after a restart, the record
+    * that failed will be reprocessed. In some use cases this pattern is more
+    * appropriate than just using auto-offset-commits, since it will not commit
+    * offsets for failed records when the consumer is closed. The consumer must
+    * be configured to disable offset auto-commits.
+    */
+  def processingAndCommitting[A](
+      pollTimeout: FiniteDuration,
+      maxRecordCount: Long = 1000L,
```

Comment:
If someone passes in a negative number, or a negative time would anything unexpected happen? I am not sure how `consumer.recordStream(-2.seconds)` would deal with it, but in the other two cases, it would mean that we would always be committing for every record. The question then becomes whether or not this is desirable and if we think that is obvious enough at the call site?

Reply:
Good point. This probably needs more validation of inputs.